### ISSUE TITLE
feat[dev]: ✨ implement A11y improvements for Badge Template

### ISSUE THREAD

**Issue by yoezlem** [solid-design-system/solid#1512]
## Description
This task describes the Storybook implementation for the Badge template as part of the A11y improvements outlined in Epic https://github.com/solid-design-system/solid/issues/1465.

A11y improvements for the Badge template can be found here: [Badge Notes](https://www.notion.so/matuzo/Badge-107f3f6e1bc0803eb6fff09eab7f5223).

-----------------------------------------

## Comment
The button with the small badge variant has no accessible name (see [@sd-badge](https://www.notion.so/matuzo/sd-badge-103f3f6e1bc080619981e3e8d9546a9f?pvs=25))

### Open Questions
- [ ] Question1
- [ ] Question2

## DoR
- [x] Item has been estimated by the team
- [x] Item dependencies have been identified and documented

## DoD
- [ ] Documentation has been created/updated (if applicable)
- [ ] Migration Guide has been created/updated (if applicable)
- [ ] Relevant E2E tests (Features, A11y, Bug fixes) are created/updated
- [ ] Relevant stories (Features, A11y) are created/updated
- [ ] Implementation works successfully on `feature` branch
